app/actions/remote/systems.ts

```diff
@@ -16,36 +15,89 @@ export type ConfigAndLicenseRequest = {
     error?: unknown;
 }
 
-export const fetchDataRetentionPolicy = async (serverUrl: string) => {
+export type DataRetentionPoliciesRequest = {
+    globalPolicy?: GlobalDataRetentionPolicy;
+    teamPolicies?: TeamDataRetentionPolicy[];
+    channelPolicies?: ChannelDataRetentionPolicy[];
+    error?: unknown;
+}
+
+export const fetchDataRetentionPolicy = async (serverUrl: string): Promise<DataRetentionPoliciesRequest> => {
+    const operator = DatabaseManager.serverDatabases[serverUrl]?.operator;
+    if (!operator) {
+        return {error: `${serverUrl} database not found`};
+    }
+
+    try {
+        const {data: globalPolicy, error: globalPolicyError} = await fetchGlobalDataRetentionPolicy(serverUrl);
+        const {data: teamPolicies, error: teamPoliciesError} = await fetchAllGranularDataRetentionPolicies(serverUrl);
+        const {data: channelPolicies, error: channelPoliciesError} = await fetchAllGranularDataRetentionPolicies(serverUrl, true);
+
+        const hasError = globalPolicyError || teamPoliciesError || channelPoliciesError;
+        if (hasError) {
+            return hasError;
+        }
+
+        return {
+            globalPolicy,
+            teamPolicies: teamPolicies as TeamDataRetentionPolicy[],
```

Comment:
Why do we need the casting here? What is the type that the fetches are returning and why we cannot set it to this one?